### PR TITLE
@gndclouds/data-bar-breakdown

### PR DESCRIPTION
**Pull request by gndclouds**
Created a new file DataMeterDeatiled which now shows an example of your total number of file and the amount of each file type
image

### CODE REVIEW

**Comment by gndclouds** [comment]
@jimmylee and @martinalong There is a CSS issue I'm having trouble finding. The DataMeterBars are stacked on top of one another (largest to smallest). But the `background-color` for each DataMeterBar is always being applied to the largest bar. I added two new props to the react fragment one for the `background-color` and one for the `z-index`.

[New props:](https://github.com/filecoin-project/slate/blob/af62ab2670a4922783baa74ab95d8954adeeda24/components/core/DataMeter.js#L199)
```html
<DataMeterBar
     bytes={props.stats.imageBytes}
     maximumBytes={props.stats.maximumBytes}
     dataMeterColor={`#C0DACD`}
     dataMeterLayer={`5`}
  />
```

[Passing props into fragment](https://github.com/filecoin-project/slate/blob/af62ab2670a4922783baa74ab95d8954adeeda24/components/core/DataMeter.js#L142): 

```html
<div
          css={STYLES_DATA_METER}
          style={{
            width: `${percentage * 100}%`,
            backgroundColor: props.dataMeterColor,
            zIndex: props.dataMeterLayer,
          }}
        ></div>
      </div>
```

I feel like I must be misunderstanding something with how react renders fragments because this should be really straight forward.

File: components/core/DataMeter.js [hunk 1]
```diff
@@ -32,13 +40,17 @@ const STYLES_DATA = css`
 
 const STYLES_DATA_METER = css`
   flex-shrink: 0;
+  position: absolute;
   height: 16px;
-  background-color: ${Constants.system.brand};
+  top: 0px;
```
Comment (jimmylee) [inline components/core/DataMeter.js:45]: I don't think the data meter needs to be an absolutely positioned component.

File: components/core/DataMeter.js [hunk 2]
```diff
@@ -83,26 +96,56 @@ const STYLES_NOTE = css`
   margin-bottom: 4px;
 `;
 
+const STYLES_DATA_METER_KEY_WRAPPER = css`
+  display: inline-block;
+`;
+const STYLES_DATA_METER_KEY_SQUARE = css`
```
Comment (jimmylee) [inline components/core/DataMeter.js:102]: New line

File: components/core/DataMeter.js [hunk 3]
```diff
@@ -83,26 +96,56 @@ const STYLES_NOTE = css`
...
+  display: inline-block;
+  border-radius: 3px;
+  background: #73ad21;
+  width: 12px;
+  height: 12px;
+  margin-right: 4px;
+  vertical-align: middle;
+`;
+const STYLES_DATA_METER_KEY_LABEL = css`
```
Comment (jimmylee) [inline components/core/DataMeter.js:111]: New line

File: components/core/DataMeter.js [hunk 4]
```diff
@@ -115,10 +158,88 @@ export const DataMeter = (props) => {
         {Strings.bytesToSize(props.stats.bytes)} of {Strings.bytesToSize(props.stats.maximumBytes)}{" "}
         used
       </div>
-      <DataMeterBar bytes={props.stats.bytes} maximumBytes={props.stats.maximumBytes} />
+
+      <div style={{ position: "relative" }}>
+        <div css={STYLES_DATA_METER_BASE}></div>
+        <DataMeterBar
```
Comment (jimmylee) [inline components/core/DataMeter.js:164]: I think the strategy should be inline-flex elements like the codepen I posted earlier: https://codepen.io/meanjim/pen/pobKKbr

**Review by jimmylee — CHANGES_REQUESTED** [review]
Would be nice for you to do the inline-flex element strategy instead. Then these elements wouldn't need to overlap.

**Comment by gndclouds** [comment]
Sorry for the confusion, I created a new component called DataMeterDetail for the Data page. Since the DataMeter is used elsewhere I did not want to alter it. This PR is not correct and uses the Flex implementation.

**Comment by jimmylee** [comment]
> This PR is not correct and uses the Flex implementation.

What does this mean?

Also I want to be clear that you can create new components. I'm just saying the `position: absolute` implementation is incorrect, take the slower pace we are moving at right now to make sure you learn the other implementation.

**Comment by jimmylee** [comment]
Branches that have new feature work should be accompanied by a screenshot or preview link as well. You can build these components without integrating with data so take that into consideration.

File: components/core/DataMeterDetailed.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  font-size: ${Constants.typescale.lvl0};
+  color: ${Constants.system.darkGray};
+  display: block;
+  margin-bottom: 4px;
+`;
+
+const STYLES_DATA_METER_KEY_WRAPPER = css`
+  display: inline-block;
+`;
+const STYLES_DATA_METER_KEY_SQUARE = css`
```
Comment (jimmylee) [inline components/core/DataMeterDetailed.js:83]: new line.

File: components/core/DataMeterDetailed.js [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  display: inline-block;
+  border-radius: 3px;
+  background: #73ad21;
+  width: 12px;
+  height: 12px;
+  margin-right: 4px;
+  vertical-align: middle;
+`;
+const STYLES_DATA_METER_KEY_LABEL = css`
```
Comment (jimmylee) [inline components/core/DataMeterDetailed.js:92]: new line

File: components/core/DataMeterDetailed.js [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+export const DataMeterBar = (props) => {
+  const percentageUsed = props.bytes / props.maximumBytes;
+  const percentageImage = props.stats.imageBytes / props.bytes;
+  const percentageVideo = props.stats.videoBytes / props.bytes;
+  const percentageEpub = props.stats.epubBytes / props.bytes;
+  const percentagePdf = props.stats.pdfBytes / props.bytes;
+  const percentageAudio = props.stats.audioBytes / props.bytes;
+  const percentageFreeSpace = props.bytes - props.maximumBytes;
+  console.log(percentageImage);
```
Comment (jimmylee) [inline components/core/DataMeterDetailed.js:110]: remove the console log

File: components/core/DataMeterDetailed.js [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+              backgroundColor: "${Constants.system.foreground}",
+            }}
+          ></div>
+        </div>
+      </div>
+    </React.Fragment>
+  );
+};
+
+export const DataMeter = (props) => {
```
Comment (jimmylee) [inline components/core/DataMeterDetailed.js:157]: DataMeterDetailed

File: components/core/DataMeterDetailed.js [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+          <div css={STYLES_DATA_METER_KEY_LABEL}>Audio</div>
+        </div>
+      </div>
+
+      <div css={STYLES_NOTE}>50GB coming soon with email verification</div>
+    </div>
+  );
+};
+
+export default DataMeter;
```
Comment (jimmylee) [inline components/core/DataMeterDetailed.js:208]: DataMeterDetailed

**Review by jimmylee — CHANGES_REQUESTED** [review]
Looks good! Tried it out, please fixes these nits and we can get this landed.

**Review by jimmylee — APPROVED** [review]
LGTM

**Pull request merged**